wp_api/src/lib.rs

```diff
@@ -212,6 +213,27 @@ pub enum IntegerOrString {
     String(String),
 }
 
+#[uniffi::export(with_foreign)]
+#[async_trait::async_trait]
+pub trait WpAppNotifier: Send + Sync + std::fmt::Debug {
+    async fn notify(&self, notification: WpAppNotification);
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, uniffi::Enum)]
+pub enum WpAppNotification {
+    UnauthorizedRequest,
+}
+
+impl WpAppNotification {
+    pub fn from_wp_network_response(response: &WpNetworkResponse) -> Option<Self> {
+        if response.status_code == 401 {
+            Some(Self::UnauthorizedRequest)
```

Comment:
For the purpose of finding out whether the current application password is still valid, only a 401 status code check is not enough, since not all 401 responses mean that an invalid application password is used.

We'll probably want to use a stricter check like

```rust
fn is_unauthorized_error(response: &WpNetworkResponse) -> bool {
    if response.status_code == 401 {
        if let Some(original_error) = WpError::try_parse(&response.body.clone()) {
            return original_error.code == WpErrorCode::Unauthorized;
        }
    }

    false
}
```

Reply:
I don't think we always get back `WpErrorCode::Unauthorized`. For example,

```
> curl --user test@example.com:invalid "http://localhost/wp-json/wp/v2/users?context=edit"
> {"code":"rest_forbidden_context","message":"Sorry, you are not allowed to list users.","data":{"status":401}}
```

So, at least for now, I have changed the condition to be true for both `WpErrorCode::Unauthorized` and `WpErrorCode::ForbiddenContext` in 80859035c89daccdf9bb3ab1982ff75ffad1aeab. Let me know what you think!

Reply:
Maybe we should check whether the HTTP response status code is 401 instead? So far, it looks like the core would always return a 401 status code, but may return a different `{"code": ...}` in the response JSON.

Reply:
The initial implementation was checking for `401`, but I tried updating it as per your previous suggestion. I agree though, I think we should only check the status code since we are going to make a request to check if the application password is valid.

Reply:
Ah right. The original approach of creating `WpAppNotification` from HTTP response probably won't work. But I see it's updated to invoke a callback after application password check. 👍